app/routes/auth.py

```
@@ -0,0 +1,271 @@
+        
+    Raises:
+        DuplicateUserError: If user with email already exists
+        WeakPasswordError: If password doesn't meet requirements
+        InternalServerError: If registration fails
+        
+    Rate Limit:
+        5 requests per minute per client
+    """
+    try:
+        if not profile_data.email or not profile_data.password:
+            raise ValidationError("Email, password, an are required fields.")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)